pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -804,6 +804,16 @@ protected void handleProducerRemoved(Producer producer) {
                 if (ex.getCause() instanceof ConsumerBusyException) {
                     log.warn("[{}][{}] Consumer {} {} already connected", topic, subscriptionName, consumerId,
                             consumerName);
+                    Consumer consumer = null;
+                    try {
+                        consumer = subscriptionFuture.isDone() ? getActiveConsumer(subscriptionFuture.get()) : null;
+                        // cleanup consumer if connection is already closed
+                        if (consumer != null && !consumer.cnx().isActive()) {
+                            consumer.close();
```

Comment:
I my understanding, this consumer should be closed elsewhere in normal situation. This means there is a chance that `consumer.close` wii be called multi times.  I think we should add closed state check in `consumer.close` to avoid other unexpected issues.

Reply:
no need to check closed state because if consumer connection is not active and if consumer is still pretends as an active consumer then it requires cleanup by closing it immedaitely.

Reply:
> if consumer connection is not active and if consumer is still pretends as an active consumer then it requires cleanup by closing it immedaitely.

I get what you are trying to do, and I am +1 for this.

But do you agree that `consumer.close` may be called in two threads at the same time? 
One is here, and the other may be in `org.apache.pulsar.broker.service.ServerCnx#channelInactive`.

Reply:
it won't happen in this code path. however, in any case it would be safe to call close for already closed consumer,

Reply:
> however, in any case it would be safe to call close for already closed consumer,

Great to know. Thx